src/Ovh/Provider.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace SocialiteProviders\Ovh;
+
+use Ovh\Api as Ovh;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+/**
+ * Class Provider.
+ */
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    public const IDENTIFIER = 'OVH';
+
+    /**
+     * @return array
+     */
+    public static function additionalConfigKeys()
+    {
+        return [
+            'endpoint',
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        $ovh = new Ovh(
```

Comment:
Generally we don't use external libraries unless we can't do plain requests. Is there a reason why you used the OVH library?

Reply:
It was to avoid reimplementing the checks and URLs. While it also uses Guzzle, it's more of a helper than anything else.
If you prefer, I can reimplement it in a class instead of an external library.

Reply:
@atymic would it be acceptable to copy paste the Api.php file from OVH library? Or do you prefer I reimplement necessary methods only?

Reply:
On seconds though, I think it's fine :)

Reply:
Then my baby girl is ready to be merged 😂